packages/flutter_tools/lib/src/ios/xcresult.dart

```diff
@@ -35,18 +36,22 @@ class XCResultGenerator {
 
   /// Generates the XCResult.
   ///
-  /// Calls `xcrun xcresulttool get --path <resultPath> --format json`,
+  /// Calls `xcrun xcresulttool get --legacy --path <resultPath> --format json`,
   /// then stores the useful information the json into an [XCResult] object.
   ///
   /// A`issueDiscarders` can be passed to discard any issues that matches the description of any [XCResultIssueDiscarder] in the list.
   Future<XCResult> generate(
       {List<XCResultIssueDiscarder> issueDiscarders =
           const <XCResultIssueDiscarder>[]}) async {
+    final Version? xcodeVersion = xcode.currentVersion;
     final RunResult result = await processUtils.run(
       <String>[
         ...xcode.xcrunCommand(),
         'xcresulttool',
         'get',
+        // See https://github.com/flutter/flutter/issues/151502
```

Comment:
can you briefly describe it in comment why this is added? so that future readers don't have to click on the link to know what this does.